### DISCUSSION TITLE
Admin panel global search

### DISCUSSION THREAD

**Discussion by susanu** [Laravel-Backpack/community-forum#1111] [Ideas]
Hi guys,

What do you think about implementing a global search that you can trigger from all pages.
Lets say that you press Shift twice and a popup would appear with a search input, as you type, results would appear grouped by the model type.

Of course, this would need to be configured by the developer, where to search and how the results would appear.

Let me know your thoughts.

**Comment by pxpm** [comment]
Love the idea. We actually have something similar in our website admin panel, but it's not polished and would require some effort to make it re-usable. 

Not sure about the shortcuts you suggested, I would expect to have like a search bar on top of the panel for example, something like this:
![image](https://github.com/user-attachments/assets/eed753f5-a95f-4ccd-bedc-dfbddd66bee2)

So maybe we need to develop like a `<x-global-search>` component that we can have some "default" places where it can appear (the top like I showed, some people may prefer to have the search on the sidebar: 
![image](https://github.com/user-attachments/assets/03364cff-0912-4470-b4e3-89fa54a6d9de)

And the developer could add that component anywhere he wants if he customized the templates, or don't like the defaults. 

Also not sure it's worth developing our own "display framework", as I think we can get way with a select2, that also give us the ajax possibility for pagination of big datasets. 

If we start by the simplest thing we can think about to have this working, we can iterate and improve on it further. 

Let me know what you think. 👍

**Reply by tabacitu** [reply]
Agreed. Global search has been a desire of mine for a long LONG time. Glad someone asked for it, maybe now we can prioritize it 😅

(and to answer Pedro, I don't think select2 would be a good fit here; I think adding a custom Livewire component is a better fit - that would provide some strong default UI, that the dev can completely configure if they want)

**Comment by susanu** [comment]
I totally agree with you.
On top of the search box on navbar, maybe we can add some kind of keyboard shortcut to activate some full page search (https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_fullscreen_search)

So when clicking the top search bar or pressing a combination of keys, this popup would appear and results will be displayed as you type.

What do you think?
